lib/SIL/Utils/MemAccessUtils.cpp

```diff
@@ -1128,22 +1132,43 @@ class FindAccessStorageVisitor
     invalidateResult();
     return SILValue();
   }
+
+  SILValue visitStorageCast(SingleValueInstruction *cast, Operand *sourceOper) {
+    auto thisCast = isAccessStorageTypeCast(cast) ? AccessStorageCast::Type
+                                                  : AccessStorageCast::Identity;
```

Comment:
> auto thisCast = isAccessStorageTypeCast(cast) ? AccessStorageCast::Type
>                                                   : AccessStorageCast::Identity;

This is fine for now. But I think it should be in the generic AccessUseDefChainVisitor

> if (isAccessStorageCast(svi))
>      return asImpl().visitStorageCast(svi, &svi->getAllOperands()[0]);

should be

```
if (isAccessStorageTypeCast(cast)) visitStorageCast(cast, oper, AccessStorageCast::Type);
if (isAccessStorageIdentityCast(cast)) visitStorageCast(cast, oper, AccessStorageCast::Identity);
```

Reply:
Fixed.